src/server/project.ts

```diff
@@ -107,6 +107,7 @@ namespace ts.server {
         private rootFilesMap: FileMap<ScriptInfo> = createFileMap<ScriptInfo>();
         private program: ts.Program;
         private externalFiles: SortedReadonlyArray<string>;
+        private missingFilesMap: FileMap<FileWatcher> = createFileMap<FileWatcher>();

```

Comment:
A `Path` is a subtype of `string`, so you can use those as map keys. (It might be slightly more type safe to specify that keys must be `Path`s, but I don't think that's worth using a `FileMap`.)